### DISCUSSION TITLE
How do I configure `.scalafmt.conf` to not indent Chisel's `when` / `.elsewhen`, etc.?

### DISCUSSION THREAD

**Discussion by pepper-lees** [scalameta/metals#5828] [Q&A]
Hi,

The default behavior of Scalameta's formatting changes chains like this:

```scala
when (foo) {
  bar := 1.U
}
.elsewhen (baz) {
  bar := 2.U
}
```
into this:
```scala
when (foo) {
  bar := 1.U
}
  .elsewhen (baz) {
    bar := 2.U
  }
```
because Chisel's `when`/`elsewhen`/`otherwise` are technically a function chain and not "real" syntax like `if`/`else if`/etc. I'd far prefer to keep them indented the same, instead of having the subsequent blocks indented.

I tried `newlines.selectChains = fold`, but I have two issues with it:

1. I'm not a fan of the hyper-condensed syntax when there's several blocks, because it makes the code hard to read:
```scala
when (foo) {
  bar := 1.U
}.elsewhen (baz) {
  bar := 2.U
}.otherwise {
  bar := 3.U
}
```
2. and if I want to add a comment above one of the blocks, it automatically indents them:
```scala
when(foo) {
  bar := 1.U
}.elsewhen(baz) {
  bar := 2.U
}
// Foo
  .otherwise {
    bar := 3.U
  }
```

Is there a way to configure .scalafmt.conf to just **not indent** function chains? I don't care if it leads to "normal" function chains being non indented, like this:
```scala
foo()
.bar()
.baz();
```
I tried using `indentOperator.excludeRegex` and `indentOperator.exclude` but nothing I did seemed to have any impact. 

Scala/Metals/Scalafmt versions:
```
$ /home/drak/chipyard/.conda-env/lib/jvm/bin/java -Xss4m -Xms100m -jar /home/drak/.cache/nvim/nvim-metals/metals --version
metals 1.1.0

# Note:
#   Supported Scala versions:
#     Scala 3: 3.1.0, 3.1.1, 3.1.2, 3.1.3, 3.2.0, 3.2.1, 3.2.2, 3.3.0, 3.3.1
#     Scala 2:
#       2.11.12
#       2.12.18, 2.12.17, 2.12.16, 2.12.15, 2.12.11, 2.12.12, 2.12.13, 2.12.14
#       2.13.12, 2.13.9, 2.13.10, 2.13.11, 2.13.5, 2.13.6, 2.13.7, 2.13.8

$ scala --version
Scala code runner version 3.3.1 -- Copyright 2002-2023, LAMP/EPFL
$ scalafmt --version
scalafmt 3.7.14
```

**Comment by tgodzik** [comment]
Thanks for the question!

I think `indentOperator.excludeRegex` will only work if you don't use a dot. There are some settings about indentation, but I don't see anything that would help here :/

**Reply by pepper-lees** [reply]
> I think indentOperator.excludeRegex will only work if you don't use a dot.

@tgodzik Are you referring to using a dot inside the regex, or are you referring to the dot inside `.elsewhen` and `.otherwise`?

**Reply by tgodzik** [reply]
inside `.elsewhen` I think the setting only applies to when we have infix notation

**Reply by pepper-lees** [reply]
Ohhh, I see. I didn't realize what an infix operator was until looking it up just now. I saw the "chain" part of _Normally, the first eligible break inside a chain of infix operators_ and thought I was looking at the right setting. 'Tis a shame if I can't get the formatter to work the way I want, but I'll settle with `newlines.selectChains = fold` because at least it'll keep the indentation. Thanks for the response!
